test_unstructured/common/test_html_table.py

```diff
@@ -33,3 +41,163 @@ def test_htmlify_matrix_handles_multiple_rows_and_cells(self):
 
     def test_htmlify_matrix_handles_empty_matrix(self):
         assert htmlify_matrix_of_cell_texts([]) == ""
+
+
+class DescribeHtmlTable:
+    """Unit-test suite for `unstructured.common.html_table.HtmlTable`."""
+
+    def it_can_construct_from_html_text(self):
+        html_table = HtmlTable.from_html_text("<table><tr><td>foobar</td></tr></table>")
+
+        assert isinstance(html_table, HtmlTable)
+        assert html_table._table.tag == "table"
+
+    @pytest.mark.parametrize(
+        "html_text",
+        [
+            "<table><tr><td>foobar</td></tr></table>",
+            "<body><table><tr><td>foobar</td></tr></table></body>",
+            "<html><body><table><tr><td>foobar</td></tr></table></body></html>",
+        ],
+    )
+    def it_can_find_a_table_wrapped_in_an_html_or_body_element(self, html_text: str):
+        html_table = HtmlTable.from_html_text(html_text)
+
+        assert isinstance(html_table, HtmlTable)
+        assert html_table._table.tag == "table"
+
+    def but_it_raises_when_no_table_element_is_present_in_the_html(self):
+        with pytest.raises(ValueError, match="`html_text` contains no `<table>` element"):
+            HtmlTable.from_html_text("<html><body><tr><td>foobar</td></tr></body></html>")
+
+    def it_removes_any_attributes_present_on_the_table_element(self):
+        html_table = HtmlTable.from_html_text(
+            '<table border="1", class="foobar"><tr><td>foobar</td></tr></table>',
+        )
+        assert html_table.html == "<table><tr><td>foobar</td></tr></table>"
+
+    @pytest.mark.parametrize(
+        "html_text",
+        [
+            "<table><thead><tr><td>foobar</td></tr></thead></table>",
+            "<table><thead><tr><td>foobar</td></tr></thead><tbody></tbody></table>",
+            "<table><tbody><tr><td>foobar</td></tr></tbody><tfoot></tfoot></table>",
+        ],
+    )
+    def it_removes_any_thead_tbody_or_tfoot_elements_present_within_the_table_element(
+        self, html_text: str
+    ):
+        html_table = HtmlTable.from_html_text(html_text)
+        assert html_table.html == "<table><tr><td>foobar</td></tr></table>"
+
+    def it_changes_any_th_elements_to_td_elements_for_cell_element_uniformity(self):
```

Comment:
This doesn't need to change, but can you explain why this behavior is desired?

Reply:
Basically, the `.text_as_html` we receive is not uniform. There are seven different partitioners that produce `.text_as_html` and they all have their own way of doing it. Some use `<td>` only, others sometimes use `<th>` etc.

Someone programming to `.text_as_html` shouldn't have to concern themselves with which partitioner originally produced the HTML, it should just be consistent. So this makes all the `.text_as_html` coming out of chunking look the same, except for the content and the number of rows and columns.

In a stacked PR I make the sources more consistent as well, but this acts as a backstop so will be belts and suspenders at that point. Also, PDF/Image is a bit of a wildcard since that's done by a model, so this will be something we'll want to leave in for the foreseeable future.